core/src/main/java/com/linecorp/armeria/internal/common/encoding/SnappyFramedOutputStream.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2023 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.internal.common.encoding;
+
+import java.io.IOException;
+import java.io.OutputStream;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.codec.compression.SnappyFrameEncoder;
+
+final class SnappyFramedOutputStream extends OutputStream {
+
+    private final SimpleSnappyFrameEncoder encoder = new SimpleSnappyFrameEncoder();
+    private final ByteBuf out;
+
+    SnappyFramedOutputStream(ByteBuf out) {
+        this.out = out;
+    }
+
+    @Override
+    public void write(int b) throws IOException {
+        // We do not optimize this since it is not used internally.
```

Comment:
Then, how about just throwing an unsupported exception so that we can efficiently implement it when this is used?